pymemcache/client/base.py

```diff
@@ -905,7 +912,12 @@ def _extract_value(self, expect_cas, line, buf, remapped_keys,
             except Exception as e:
                 raise ValueError("Unable to parse line %s: %s" % (line, e))
 
-        buf, value = _readvalue(self.sock, buf, int(size))
+        value = None
+        try:
+            buf, value = _readvalue(self.sock, buf, int(size))
+        except MemcacheUnexpectedCloseError as e:
+            self.close()
+            raise e
```

Comment:
I think we just want to `raise` here (without capturing and re-raising `e`). Below, too.